toptal/picasso
# chore(AvatarUpload): add style prop for inner Avatar component

[FX-3158]

### Description

According to the ticket, users want to add their own style for the inner `Avatar` component. The reason is that they want to crop the image when they need. So, I introduced a new prop `avatarStyle` to pass it to `Avatar` directly

### How to test

- You can go to https://picasso.toptal.net/fx-3158-add-style-prop-for-avatarupload/?path=/story/components-avatarupload--avatarupload#with-upload and add `avatarStyle` prop and play with it. 
- Don't forget to upload an avatar first to see your result.

### Screenshots

<img width="540" alt="Screen Shot 2022-10-11 at 10 45 57" src="https://user-images.githubusercontent.com/7733167/195030328-5326e65d-7354-4209-8020-17bbcf64272b.png">

### Development checks

- Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- Make sure that additions and changes on design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- Create `examples` for component
- Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- Covered with tests

**Breaking change**

- codemod is created and showcased in the changeset
- test alpha package of Picasso in StaffPortal

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>
